store.rb

```diff
@@ -35,14 +59,25 @@ def session_id
   def stop()
     @log.warn "stopping the store"
     Process.kill("TERM", Process.pid) unless $EXIT
+    @root_watcher.unsubscribe unless @root_watcher.nil?
+    Thread.kill(@fetch_thread) unless @fetch_thread.nil?
     @zk.close() if @zk
   end
 
+  # get instances for a given service
   def nodes()
+    return load_instances_from_zk unless @use_cache
+    instances = get_cached_instances()
+    instances ||= {}
```

Comment:
Not sure if this is ruby's idiom, but in JavaScript you merge the two lines as `instances = get_cached_instances() || {}`